Hydra/Serializer/ConstraintViolationListNormalizer.php

```diff
@@ -41,13 +41,16 @@ public function normalize($object, $format = null, array $context = [])
         if ($object instanceof ConstraintViolationListInterface) {
             $message = '';
 
-            foreach ($object as $violation) {
+            foreach ($object as $key => $violation) {
                 $violations[] = [
                     'propertyPath' => $violation->getPropertyPath(),
                     'message' => $violation->getMessage(),
                 ];
 
-                $message .= ($violation->getPropertyPath() ? $violation->getPropertyPath().': ' : '').$violation->getMessage()."\n";
+                $message .= ($violation->getPropertyPath() ? $violation->getPropertyPath().': ' : '').$violation->getMessage();
+                if ($key < count($object) - 1) {
```

Comment:
Or we could just have a `$messages` array and then `$message = implode("\n", $messages)`...